includes/sanitizers/class-amp-link-sanitizer.php

```diff
@@ -132,12 +150,26 @@ public function process_links() {
 			}
 
 			$href = $element->getAttribute( 'href' );
-
-			if ( $this->is_frontend_url( $href ) && '#' !== substr( $href, 0, 1 ) ) {
+			$rel  = $element->hasAttribute( 'rel' ) ? array_filter( preg_split( '/\s+/', $element->getAttribute( 'rel' ) ) ) : [];
+			$pos  = array_search( self::REL_VALUE_NON_AMP_TO_AMP, $rel, true );
+			if ( false !== $pos ) {
+				// The rel has a value to opt-out of AMP-to-AMP links, so strip it and ensure the link is to non-AMP.
+				unset( $rel[ $pos ] );
+				if ( empty( $rel ) ) {
+					$element->removeAttribute( 'rel' );
+				} else {
+					$element->setAttribute( 'rel', implode( ' ', $rel ) );
+				}
+			} elseif (
+				$this->is_frontend_url( $href )
+				&&
+				'#' !== substr( $href, 0, 1 )
+				&&
+				! in_array( $href, $this->args['excluded_amp_links'], true )
```

Comment:
Any fragment identifier should be omitted from consideration here.

```suggestion
				! in_array( strtok( $href, '#' ), $this->args['excluded_amp_links'], true )
```

Reply:
Sure, https://github.com/ampproject/amp-wp/pull/4146/commits/f85e8be23179001e348fd2a2dd8d0639a73a411f applies that change, and adds a test case for it.